XpringKit/ILP/auth/IlpCredentials.swift

```diff
@@ -0,0 +1,33 @@
+import SwiftGRPC
+
+/// A helper which provides a convenient way to initialize grpc.Metadata and
+/// add an Authorization metadata header, and ensures every bearer token
+/// going over the wire is prefixed with "Bearer "
+class IlpCredentials {
+
+    /// Unfortunately, Metadata is not an open class in SwiftGRPC, so we cannot extend it.
+    /// Instead, it is wrapped in IlpCredentials
+    private var metadata: Metadata
+
+    /// Initialize a new IlpCredentials
+    /// self.metadata will be initialized and an Authorization header will be added to it.  The value
+    /// of that entry will be the bearerToken with "Bearer " prefix.
+    public init(_ bearerToken: String) throws {
+        self.metadata = Metadata()
+        try metadata.add(key: "authorization", value: self.applyBearer(bearerToken))
+    }
+
+    /// Prepends 'Bearer ' to an auth token, if it is not already prefixed with "Bearer "
+    ///
+    /// - Parameters:
+    ///     - token: An auth token that either does or does not have a "Bearer " prefix
+    /// - Returns: A bearer token as a String, with a "Bearer " prefix
+    private func applyBearer(_ token: String) -> String {
+        return token.starts(with: "Bearer ") ? token : "Bearer " + token
+    }
+
+    /// Get the Metadata to pass to network calls
+    public func get() -> Metadata {
```

Comment:
Is `MetaData` a struct or a class? If it's a struct then it's copy on assign and you can just make your variable public. 

If it isn't I'd recommend using `getMetadata` or something? `get()` is only descriptive if you know the class structure, IMO

Reply:
`Metadata` is a class.  I considered doing something like `getMetadata`, but ended up going with `get()` since `metadata` is the only thing we're wrapping here and `get()` just rolls off the tongue a little better.  I'll change to `getMetadata` though, since it's clearer and I'm not super worried about being pretty in internal stuff.

Reply:
wow, bummer on the class. This sounds reasonable though.